lib/rules/no-inner-declarations.js

```diff
@@ -97,8 +111,24 @@ module.exports = {
 
         return {
 
-            FunctionDeclaration: check,
+            FunctionDeclaration(node) {
+                const isStrictFunction = sourceCode.getScope(node).isStrict;
+
+                if (isStrictFunction && legacy === false) {
+                    return;
+                }
+
+                if (isStrictFunction && ecmaVersion !== 5) {
+                    return;
+                }
+
+                check(node);
+            },
             VariableDeclaration(node) {
+                if (sourceCode.getScope(node).isStrict) {
+                    return;
+                }
```

Comment:
I don't think we should change anything regarding variable declarations, because they behave the same in all contexts.

Sorry for the lack of precise design details in https://github.com/eslint/eslint/issues/15576. Since we plan to remove this rule from the recommended configuration in v9.0.0-alpha.0 which will be released today, I prepared a separate PR just for that (https://github.com/eslint/eslint/pull/17920), and will focus on this issue next week.

Reply:
is this means we should not check variable declaration if it is on non-strict or strict mode and keep it as it is right now?